README.md

```diff
@@ -112,3 +93,27 @@ spec:
         - service: sample-svc-asia1
           weight: 0
 ```
+
+## Implementation details
+
+### Failover criteria
+
+The failover criteria is readiness failures on the targeted Pods. This is
+directly reflected on the Endpoints pointing to those Pods: only when Pods are
+ready, does the `addresses` field of the relevant Endpoints get populated.
+
+### Failover logic
+
+The following describes the logic used to change the `TrafficSplit` weights:
+
+- Whenever the primary backend is ready, all the weight is set to it, setting
+  the weights for all the secondary backends to zero.
+- Whenever the primary backend is not ready, the following rules apply only if
+  there is at least one secondary backend that is ready:
+  - The primary backend’s weight is set to zero.
+  - The weight is distributed equally among all the secondary backends that
+    are ready.
+  - Whenever a secondary backend changes its readiness, the weight is
+    redistributed among all the secondary backends that are ready
+- Whenever both the primary and secondaries are all unavailable, the connection
```

Comment:
TIOLI: we could remove `all` from `all unavailable` perhaps, since we already say "both the primary and secondaries". I don't think the current way makes it hard to understand so I'm definitely fine to leave it as is.

Reply:
:+1: